src/main/kotlin/com/hibob/kotlinBasics/types/StoreService.kt

```diff
@@ -0,0 +1,44 @@
+package com.hibob.academy.types
+
+import com.hibob.kotlinBasics.types.*
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+       return cart.map({it -> it.clientId to checkout(it, payment)}).toMap()
```

Comment:
few things here:
1. instead of using map p, you can use associate function that convert it to map.
2. if you use -> don't call the parameter it, it is usually our wat to reach the current object without the -> meaning you can choose between:
`return cart.associate { it.clientId to checkout(it, payment) }  `
or
`        return cart.associate { item -> item.clientId to checkout(item, payment) }
`
in the second example is like your implementation with renam of it to item

3. don't need enter in the end of the function
4. this is the only method that should be public all the rest can be private